File: tangenteHiperbolica.m
% Ian Mu;oz Nu;ez - Funcion tangente hiperbolica

close all
clear
clc

n = 1000;

v = linspace(-10, 10, n);

A = 1; % Amplifica la funcion
B = 0.2; % Modifica la pendiente de la funcion
C = 0; % Desfasa la funcion

phi = zeros(5, n);
for i=1:5
    for j=1:n
        phi(i, j) = A * tanh(B*v(j) + C);
    end
    B = B + 0.2;
end

figure(1)
hold on
grid on

plot(v, phi(1, :), 'r', 'LineWidth', 2)
plot(v, phi(2, :), 'b', 'LineWidth', 2)
plot(v, phi(3, :), 'g', 'LineWidth', 2)
plot(v, phi(4, :), 'y', 'LineWidth', 2)
plot(v, phi(5, :), 'm', 'LineWidth', 2)

title("Funcion tangente hiperbolica", 'FontSize', 20)
xlabel('v', 'FontSize', 15)
ylabel('\phi(v)', 'FontSize', 15)
legend('0.2', '0.4', '0.6', '0.8', '1')

File: tangenteHiperbolica.py
# Ian Mu;oz Nu;ez - Funcion tangente hiperbolica

import matplotlib.pyplot as plt
import numpy as np

n = 1000

v = np.linspace(-10, 10, n)

A = 1 # Amplifica la funcion
B = 0.2 # Modifica la pendiente de la funcion
C = 0 # Desfasa la funcion

phi = np.zeros((5, n))
for i in range(0, 5):
    for j in range(0, n):
        phi[i, j] = A * np.tanh(B*v[j] + C)
    B += 0.2

plt.figure(1)
plt.grid()

plt.plot(v, phi[0, :], 'r', linewidth=2, label=r'0.2')
plt.plot(v, phi[1, :], 'b', linewidth=2, label=r'0.4')
plt.plot(v, phi[2, :], 'g', linewidth=2, label=r'0.6')
plt.plot(v, phi[3, :], 'y', linewidth=2, label=r'0.8')
plt.plot(v, phi[4, :], 'm', linewidth=2, label=r'1')

plt.title("Funcion tangente hiperbolica", fontsize=20)
plt.xlabel('v', fontsize=15)
plt.ylabel('$\phi$(v)', fontsize=15)
plt.legend()
plt.show()

